--- a/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c	1970-01-01 08:00:01.000000000 +0800
+++ b/drivers/pinctrl/rockchip/pinctrl-rockchip-core.c	2023-06-14 01:56:23.364063607 +0800
@@ -40,7 +40,7 @@
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_recalced_data *data;
+	struct rockchip_mux_recalced_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_recalced; i++) {
@@ -64,7 +64,7 @@
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_route_data *data;
+	struct rockchip_mux_route_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_routes; i++) {
@@ -423,14 +423,14 @@
 	u32 cells[MAX_ROCKCHIP_PINS_ENTRIES * 4];
 	u32 bank, pin, mux, conf, arg, default_val;
 	int ret, count, i;
-	const char *prop_name;
-	const void *value;
+	const char *prop_name = NULL;
+	const void *value = NULL;
 	int prop_len, param;
-	const u32 *data;
+	const u32 *data = NULL;
 	ofnode node;
 #if CONFIG_IS_ENABLED(OF_LIVE)
-	const struct device_node *np;
-	struct property *pp;
+	const struct device_node *np = NULL;
+	struct property *pp = NULL;
 #else
 	int property_offset, pcfg_node;
 	const void *blob = gd->fdt_blob;
--- a/drivers/pinctrl/pinctrl-rockchip.c	1970-01-01 08:00:01.000000000 +0800
+++ b/drivers/pinctrl/pinctrl-rockchip.c	2023-06-14 01:59:46.814286888 +0800
@@ -829,7 +829,7 @@
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_recalced_data *data;
+	struct rockchip_mux_recalced_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_recalced; i++) {
@@ -1561,7 +1561,7 @@
 {
 	struct rockchip_pinctrl_priv *priv = bank->priv;
 	struct rockchip_pin_ctrl *ctrl = priv->ctrl;
-	struct rockchip_mux_route_data *data;
+	struct rockchip_mux_route_data *data = NULL;
 	int i;
 
 	for (i = 0; i < ctrl->niomux_routes; i++) {
@@ -2940,14 +2940,14 @@
 	u32 cells[MAX_ROCKCHIP_PINS_ENTRIES * 4];
 	u32 bank, pin, mux, conf, arg, default_val;
 	int ret, count, i;
-	const char *prop_name;
-	const void *value;
+	const char *prop_name = NULL;
+	const void *value = NULL;
 	int prop_len, param;
-	const u32 *data;
+	const u32 *data = NULL;
 	ofnode node;
 #if defined(CONFIG_OF_LIVE) && !defined(CONFIG_SPL_BUILD)
-	const struct device_node *np;
-	struct property *pp;
+	const struct device_node *np = NULL;
+	struct property *pp = NULL;
 #else
 	int property_offset, pcfg_node;
 	const void *blob = gd->fdt_blob;
